src/main/java/com/google/api/codegen/metacode/FieldStructureParser.java

```diff
@@ -64,54 +64,70 @@ private static InitValueConfig createInitValueConfig(
     return valueConfig;
   }
 
+  // config:
```

Comment:
This is so natural to me now that I have a hard time clarifying it :/. This is the grammar for config written in the form of similar to `yacc`. Think of it as a recursive function.

```
config:
  ident
  config '.' ident
```
is essentially saying config is an identifier or a config allowed by a dot and an identifier. So `a` is a valid config (identifier), `a.b` is a config because `a` is a config followed by a dot, followed by identifier `b`.

Reply:
My initial confusion came from the fact that the formal parameter `config` is not the same as the `config` you have in the comments: the function itself is not recursive like the grammar is, so we should not refer to the grammar element and the actual parameter by the same name.

I suggest changing the parameter name to `path` and changing the comment to something like this:
```
 // Parses `path` to construct the `InitCodeNode` it specifies. `path` must be a valid config satisfying the recursive grammar below:
  // config:
  //   ident
  //   config '.' ident
  //   config '[' number ']'
  //   config '{' number '}'
  //   config '{' string '}'
  //   config '{' ident '}' (for compatibility, the identifier is just treated as a string)
private static InitCodeNode parsePartialDottedPathToInitCodeNode(
      String path, InitValueConfig initValueConfig) {
```